app/controllers/CheckYourAnswersController.scala

```diff
@@ -87,38 +89,28 @@ class CheckYourAnswersController @Inject() (
-     request.userAnswers.get(DoYouHaveUniqueTaxPayerReferencePage),
-     request.userAnswers.get(BusinessTypePage)
-    ) match { //ToDo defaulting to registering for business change when paths created if necessary
-      case (None, Some(true), Some(Sole))                   => false
-      case (None, Some(true), _)                            => true
-      case (Some(RegistrationTypeBusiness), Some(false), _) => true
-      case _                                                => false
-    }
-
-  private def createSubscription(regime: Regime, userAnswers: UserAnswers)(implicit request: DataRequest[AnyContent], hc: HeaderCarrier): Future[Result] =
-    subscriptionService.createSubscription(userAnswers) flatMap {
-      case Right(subscriptionID) =>
-        logger.info(s"The subscriptionId id $subscriptionID")
-        Future.successful(NotImplemented("Not implemented"))
-      case Left(MandatoryInformationMissingError) => Future.successful(Redirect(routes.SomeInformationIsMissingController.onPageLoad(regime)))
-      case Left(DuplicateSubmissionError) =>
-        Future.successful(NotImplemented("DuplicateSubmission is not implemented")) //TODO create OrganisationHasAlreadyBeenRegistered page
-      case Left(BadRequestError) => renderer.render("thereIsAProblem.njk").map(BadRequest(_))
-      case _                     => renderer.render("thereIsAProblem.njk").map(InternalServerError(_))
-    }
-
   def onSubmit(regime: Regime): Action[AnyContent] = (identify(regime) andThen getData.apply andThen requireData(regime)).async {
     implicit request =>
-      (request.userAnswers.get(DoYouHaveUniqueTaxPayerReferencePage),
-       request.userAnswers.get(WhatAreYouRegisteringAsPage),
-       request.userAnswers.get(DoYouHaveNINPage)
-      ) match {
-        case (Some(false), _, Some(false) | None) => Future.successful(NotImplemented("Not implemented")) // TODO DAC6-1142
-        case (Some(_), _, Some(true) | None)      => createSubscription(regime, request.userAnswers)
-        case _                                    => Future.successful(Redirect(routes.SomeInformationIsMissingController.onPageLoad(regime)))
-      }
+      (for {
+        registrationInfo   <- getEither(RegistrationInfoPage).orElse(EitherT(registrationService.registerWithoutId))
+        subscriptionID     <- EitherT(subscriptionService.createSubscription(registrationInfo.safeId, request.userAnswers))
+        withSubscriptionID <- setEither(SubscriptionIDPage, SubscriptionID(subscriptionID))
+        _ = sessionRepository.set(withSubscriptionID)
+        _ <- EitherT(taxEnrolmentsService.createEnrolment(registrationInfo.safeId, withSubscriptionID, subscriptionID, regime))
+      } yield Redirect(routes.RegistrationConfirmationController.onPageLoad(regime)))
+        .valueOrF {
+          case MandatoryInformationMissingError(error) =>
+            Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad(regime, None)))
```

Comment:
@iservport Instead of JourneyrEcovery it should go to 

```suggestion
            case Left(MandatoryInformationMissingError) => Future.successful(Redirect(routes.SomeInformationIsMissingController.onPageLoad(regime)))
```